packages/react/src/primitives/Menu/__tests__/menu.test.tsx

```diff
@@ -161,5 +207,85 @@ describe('Menu', () => {
 
       expect(disabled).toHaveClass('amplify-button--disabled');
     });
+
+    it('should add the Amplify UI Button disabled class to when Menu is disabled', async () => {
+      render(
+        <Menu isOpen isDisabled>
+          {/* Force open to test menu items */}
+          <MenuItem>Option 1</MenuItem>
+          <MenuItem>Option 2</MenuItem>
+          <MenuItem testId="disabled_option">Option 3</MenuItem>
+        </Menu>
+      );
+
+      const disabled = await screen.findByTestId('disabled_option');
+
+      expect(disabled).toHaveClass('amplify-button--disabled');
+    });
+
+    it('should behave as expected when Menu is disabled', async () => {
+      const clickHandler = jest.fn();
+      const { rerender } = render(
+        <Menu isOpen isDisabled={false}>
+          <MenuItem>Option 1</MenuItem>
+          <MenuItem>Option 2</MenuItem>
+          <MenuItem testId="enabled_menu_item" onClick={clickHandler}>
+            Option 3
+          </MenuItem>
+        </Menu>
+      );
+
+      const menuItem = await screen.findByTestId('enabled_menu_item');
+      expect(menuItem).not.toHaveAttribute('disabled');
+      expect(menuItem).not.toHaveClass('amplify-button--disabled');
+
+      act(() => {
+        menuItem.dispatchEvent(
+          new KeyboardEvent('keydown', { bubbles: true, key: 'Enter' })
+        );
+      });
+
+      expect(clickHandler).toHaveBeenCalledTimes(1);
+
+      rerender(
+        <Menu isOpen isDisabled>
+          <MenuItem>Option 1</MenuItem>
+          <MenuItem>Option 2</MenuItem>
+          <MenuItem testId="enabled_menu_item" onClick={clickHandler}>
+            Option 3
+          </MenuItem>
+        </Menu>
+      );
+
+      expect(menuItem).toHaveAttribute('disabled');
+      expect(menuItem).toHaveClass('amplify-button--disabled');
+
+      act(() => {
+        menuItem.dispatchEvent(
+          new KeyboardEvent('keydown', { bubbles: true, key: 'Enter' })
+        );
+      });
+
+      expect(clickHandler).toHaveBeenCalledTimes(1); // Should not be called an additional time
+    });
+  });
+
+  describe('menu button', () => {
+    it('should add the Amplify UI Button disabled class to disabled MenuButton', async () => {
```

Comment:
We can simplify the test description here to only include the behavioral assertion:
```suggestion
    it('has the expected class when disabled', async () => {
```